pkg/chunk/cache/memcached_client.go

```diff
@@ -98,8 +98,10 @@ func NewMemcachedClient(cfg MemcachedClientConfig, name string) MemcachedClient
 		hostname:   cfg.Host,
 		service:    cfg.Service,
 		addresses:  strings.Split(cfg.Addresses, ","),
-		provider:   dns.NewProvider(util.Logger, prometheus.DefaultRegisterer, dns.GolangResolverType),
-		quit:       make(chan struct{}),
+		provider: dns.NewProvider(util.Logger, prometheus.WrapRegistererWith(prometheus.Labels{
```

Comment:
Can we supply registrerer to `NewMemcachedClient` function instead of using `prometheus.DefaultRegisterer`? We're moving to that approach across the codebase, this PR seems like a nice opportunity to do it here.